routers/repo/pull_review.go

```diff
@@ -61,6 +61,56 @@ func CreateCodeComment(ctx *context.Context, form auth.CodeCommentForm) {
 	ctx.Redirect(comment.HTMLURL())
 }
 
+// UpdateResolveConversation add or remove an Conversation resolved mark
+func UpdateResolveConversation(ctx *context.Context) {
+	action := ctx.Query("action")
+	issueID := ctx.QueryInt64("issue_id")
+	commentID := ctx.QueryInt64("comment_id")
+
+	issue, err := models.GetIssueByID(issueID)
+	if err != nil {
+		ctx.ServerError("GetIssueByID", err)
+		return
+	}
+
+	var permResult bool
+	if permResult, err = models.CanMarkConversation(issue, ctx.User); err != nil {
+		ctx.ServerError("CanMarkConversation", err)
+		return
+	}
+	if !permResult {
+		ctx.ServerError("UpdateResolveConversation",
```

Comment:
This should probably be a "forbidden" or at least "bad request" (4xx), not a "server error" (5xx).

Reply:
I'm sorry , but I don't know how to do it.  do you means this way? 

```Go
ctx.JSON(400, map[string]interface{}{
		// some error message ...
	})
```
Thanks.

Reply:
This would suffice:
```
ctx.Error(403)
```
Unless we intend to parse the response on the client, in which case I guess `ctx.JSON(403, ...` would be appropriate; but I don't think we need to do the parsing.